src/Agent/NewRelic/Agent/Extensions/Providers/Wrapper/NServiceBus/LoadHandlersConnectorWrapper.cs

```diff
@@ -55,14 +55,23 @@ public CanWrapResponse CanWrap(InstrumentedMethodInfo methodInfo)
 
             void OnComplete(Task task)
             {
-                if (task?.Status == TaskStatus.RanToCompletion)
+                if (task == null)
                 {
-                    transaction.End();
```

Comment:
Do we want to end the segment to?

Reply:
I'm not sure when/how OnComplete is called with a null Task, but doing nothing if it is null was copied from some default delegate implementations. Unless you're sure we need to end the segment, then I think we should leave it out here

Reply:
nah, that makes sense.